echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py

```diff
@@ -0,0 +1,1091 @@
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    # Map Beam --> Sonar/Beam_group1
+    if "Beam" in ed_obj.group_paths:
+        ed_obj._tree["Sonar"].add_child(ed_obj._tree["Beam"])
+        ed_obj._tree["Sonar/Beam"].name = "Beam_group1"
+
+    # Map Beam_power --> Sonar/Beam_group2
+    if "Beam_power" in ed_obj.group_paths:
+        ed_obj._tree["Sonar"].add_child(ed_obj._tree["Beam_power"])
+        ed_obj._tree["Sonar/Beam_power"].name = "Beam_group2"
+
+
+def get_channel_id(ed_obj, sensor):
+    """
+    Returns the channel_id for all non-unique frequencies.
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x.
+    sensor : str
+        The sensor used to create the v0.5.x file.
+
+    Returns
+    -------
+    A datarray specifying the channel_ids with dimension frequency.
+    """
+    if sensor == "AZFP":
+
+        # create frequency_nominal variable
+        freq_nom = ed_obj["Sonar/Beam_group1"].frequency
+
+        # create unique channel_id for AZFP
+        freq_as_str = (freq_nom / 1000.0).astype(int).astype(str).values
+        channel_id_str = [
+            str(ed_obj["Sonar"].sonar_serial_number) + "-" + freq_as_str[i] + "-" + str(i + 1)
+            for i in range(len(freq_as_str))
+        ]
+        channel_id = xr.DataArray(
+            data=channel_id_str, dims=["frequency"], coords={"frequency": freq_nom}
+        )
+
+    else:
+
+        channel_id = xr.concat(
+            [child.ds.channel_id for child in ed_obj._tree["Sonar"].children], dim="frequency"
+        )
+
+    return channel_id
+
+
+def _frequency_to_channel(ed_obj, sensor):
+    """
+    1. In all groups that it appears, changes the dimension
+    ``frequency`` to ``channel`` whose values are based on
+    ``channel_id`` for EK60/EK80 and are a custom string
+    for AZFP.
+    2. Removes channel_id if it appears as a variable
+    3. Adds the variable ``frequency_nominal`` to all
+    Datasets that have dimension ``channel``
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x.
+    sensor : str
+        The sensor used to create the v0.5.x file.
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    channel_id = get_channel_id(ed_obj, sensor)  # all channel ids
+
+    for grp_path in ed_obj.group_paths:
+
+        if "frequency" in ed_obj[grp_path]:
+
+            # add frequency_nominal, remove standard_name attribute, and
+            # rename frequency to channel
+            ed_obj[grp_path]["frequency_nominal"] = ed_obj[grp_path].frequency
+
+            if "standard_name" in ed_obj[grp_path].frequency_nominal.attrs:
+                del ed_obj[grp_path].frequency_nominal.attrs["standard_name"]
```

Comment:
I think we can keep this, since the nominal frequency _is_ sound frequency.

Reply:
Yes, keep
```suggestion
```

Reply:
This attribute was dropped when we created this variable. So, we need to do a PR that adds this attribute back in (occurs in `set_groups_X.py`). If we all agree that this should be included, is the following attribute appropriate (this is what it was): 

`frequency_nominal.attrs["standard_name"] = "sound_frequency"`

Reply:
Yes, please go with that. You can self-merge that PR to speed this up.

Reply:
yes